### PR TITLE
Add Adafruit Feather ESP32-C6

### PR DESCRIPTION
**Pull request by tyeth**

### CODE REVIEW

**Comment by brentru** [comment]
@tyeth  The CI is encountering a failure for the c6, https://github.com/adafruit/Adafruit_Wippersnapper_Arduino/actions/runs/10371737786/job/28712750874?pr=615

**Comment by tyeth** [comment]
Yep, needs the boards PR merged to pickup the offset from the boards json

File: platformio.ini [hunk 1]
```diff
@@ -142,6 +142,32 @@ board = adafruit_itsybitsy_esp32
...
 
+; Adafruit ESP32 Feather C6
+[env:adafruit_feather_esp32c6_4mbflash_nopsram]
+extends = common:esp32
+board = adafruit_feather_esp32c6
+build_flags = -DARDUINO_ADAFRUIT_FEATHER_ESP32C6
+board_build.filesystem = littlefs
+board_build.partitions = min_spiffs.csv
+
+; Espressif ESP32-C6 4MB NO PSRAM esp32-c6-devkitm-1
```
Comment (brentru) [inline platformio.ini:153]: Do we need this? Are you hoping to add the c6 devkit?

**Review by brentru — CHANGES_REQUESTED** [review]

File: platformio.ini (hunk 1 above)
Comment (tyeth) [reply platformio.ini:153]: I've been adding and using the Espressif devkit versions just for debugging, only the S3 and this C6 so far, but long term the plan was to add them as supported boards.
 It felt like doing all the Adafruit boards first was best, and I think it was mentioned in a meeting to do the LoLin and a couple of other boards next, then the esp devkits. Think that was based on popularity.
I've had a quick search of basecamp, but can't see anything Lolin / new boards list related, it must have been a discussion in our weekly meeting before the summer.

Comment (brentru) [reply platformio.ini:153]: Okay!

**Review by brentru — APPROVED** [review]
@tyeth LGTM

**Pull request merged**